substrate/frame/ranked-collective/src/lib.rs

```diff
@@ -650,6 +656,44 @@ pub mod pallet {
 				pays_fee: Pays::No,
 			})
 		}
+
+		/// Exchanges a member with a new account and the same existing rank.
+		///
+		/// - `origin`: Must be the `AdminOrigin`.
+		/// - `who`: Account of existing member of rank greater than zero to be exchanged.
+		/// - `new_who`: New Account of existing member of rank greater than zero to exchanged to.
+		/// - `min_rank`: The rank of the member or greater.
+		///
+		/// Weight: `O(min_rank)`.
+		#[pallet::call_index(6)]
+		#[pallet::weight(T::WeightInfo::exchange_member())]
+		pub fn exchange_member(
+			origin: OriginFor<T>,
+			who: AccountIdLookupOf<T>,
+			new_who: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			// remove who
+			let max_rank = T::ExchangeOrigin::ensure_origin(origin)?;
+			let who = T::Lookup::lookup(who)?;
+			let MemberRecord { rank, .. } = Self::ensure_member(&who)?;
+
+			for r in 0..=rank {
+				Self::remove_from_rank(&who, r)?;
+			}
+			Members::<T, I>::remove(&who);
+
+			// add new member
+			let new_who = T::Lookup::lookup(new_who)?;
+			Self::do_add_member(new_who.clone())?;
```

Comment:
You can use `do_add_member_to_rank`. But it would emit `RankChanged` events, which is not really correct...

Reply:
Good idea, we could just add some boolean for whether we should emit an event or not.